### PR TITLE
GetGuardianData

### PR DESCRIPTION
**Pull request by sstanculeanu**

### CODE REVIEW

File: data/guardian.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+type GuardianDataResponse struct {
+	Data struct {
+		GuardianData *GuardianData `json:"guardianData"`
+	} `json:"data"`
+	Error string `json:"error"`
+	Code  string `json:"code"`
+}
+
+// Guardian holds the relevant information for an account guardian
+type Guardian struct {
```
Comment (AdoAdoAdo) [inline data/guardian.go:13]: can you import the GuardianData and Guardian structures from elrond-go-core instead?

File: go.mod [hunk 2]
```diff
@@ -3,8 +3,8 @@ module github.com/ElrondNetwork/elrond-sdk-erdgo
 go 1.14
 
 require (
-	github.com/ElrondNetwork/elrond-go v1.3.38-0.20220901083346-925f26245a35
-	github.com/ElrondNetwork/elrond-go-core v1.1.20-0.20220901061429-fc1143a4a107
+	github.com/ElrondNetwork/elrond-go v1.3.38-0.20220912122303-9c2574322163
```
Comment (AdoAdoAdo) [inline go.mod:6]: :+1:

**Review by AdoAdoAdo — CHANGES_REQUESTED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

File: blockchain/proxy.go [hunk 3]
```diff
@@ -539,6 +540,37 @@ func (ep *elrondProxy) GetGenesisNodesPubKeys(ctx context.Context) (*data.Genesi
...
+	err := ep.checkFinalState(ctx, address.AddressAsBech32String())
+	if err != nil {
+		return nil, err
+	}
+	if check.IfNil(address) {
+		return nil, ErrNilAddress
+	}
+	if !address.IsValid() {
+		return nil, ErrInvalidAddress
+	}
```
Comment (BeniaminDrasovean) [inline blockchain/proxy.go:554]: Move this before L545. And add test for nil address.

**Review by BeniaminDrasovean — CHANGES_REQUESTED** [review]

File: blockchain/proxy.go (hunk 3 above)
Comment (sstanculeanu) [reply blockchain/proxy.go:554]: thanks, fixed GetAccount as well

**Review by AdoAdoAdo — APPROVED** [review]

**Review by BeniaminDrasovean — APPROVED** [review]
